src/js/actions/ui.js

```diff
@@ -99,6 +100,9 @@ define(function (require, exports) {
                 };
 
                 this.dispatch(events.ui.TRANSFORM_UPDATED, payload);
+            })
+            .then(function () {
+                this.flux.actions.tools.resetBorderPolicies();
```

Comment:
Ditto: can be moved into the previous handler, and the promise should be returned.